test/apps/iodemo/io_demo.cc

```diff
@@ -302,22 +339,36 @@ class P2pDemoCommon : public UcxContext {
             return tr_hdr()->sn;
         }
 
-        uint16_t chksum() const {
-            return chk_hdr()->chk_sum;
+        uint32_t seed() const {
+            return chk_hdr()->seed;
         };
 
-        inline uint16_t calc_chksum() const {
+        inline size_t validate() const {
             if (_iov.size() == 0) {
                 return 0;
             }
 
-            uint16_t chk_sum = P2pDemoCommon::calc_chksum(_iov[0]->buffer(),
-                                                          _iov[0]->size());
+            unsigned iov_seed  = seed();
+            void *buf          = tr_hdr() + 1;
+            size_t size        = _iov[0]->size() - (sizeof(chk_hdr_t) +
+                                                    sizeof(tr_hdr_t));
+            size_t err_pos     = IoDemoRandom::validate(iov_seed, buf, size);
+            if (err_pos < size) {
+                return err_pos;
+            }
+
+            size_t pos = err_pos;
             for (size_t i = 1; i < _iov.size(); ++i) {
-                chk_sum ^= ucs_crc16(_iov[i]->buffer(), _iov[i]->size());
+                buf     = _iov[i]->buffer();
+                size    = _iov[i]->size();
+                err_pos = IoDemoRandom::validate(iov_seed, buf, size);
+                pos    += err_pos;
```

Comment:
move after line 368: pos += size

Reply:
it should be here to calculate "total" err position, `err_pos` is an error position in `_iov[i]`